examples/shadowmap/main.js

```diff
@@ -1267,12 +1289,47 @@
             // need camera position in world too
             this._config[ 'camera' ] = this._viewer.getCamera();
 
-            var numLights = this._config.lightNum;
+            var numLights = ~~this._config.lightNum;
             var lightScale = 1.0 / numLights - 1e-4;
 
 
-            for ( var k = 0; k < numLights; k++ ) {
-                this.addShadowedLight( group, k, lightScale );
+            if ( this._config[ 'atlas' ] ) {
+
+                for ( var k = 0; k < numLights; k++ ) {
+
+                    var shadowSettings = new osgShadow.ShadowSettings( this._config );
+                    shadowSettings.atlasSize = 2048;
+                    var mapres = parseInt( this._config[ 'textureSize' ] );
+                    shadowSettings.setTextureSize( mapres );
+                    shadowSettings.setCastsShadowDrawTraversalMask( this._castsShadowDrawTraversalMask );
+                    shadowSettings.setCastsShadowBoundsTraversalMask( this._castsShadowBoundsTraversalMask );
+                    this._shadowSettings[ k ] = shadowSettings;
+
+                }
+
+                var shadowMapAtlas = new osgShadow.ShadowMapAtlas( this._shadowSettings[ 0 ] );
+                this._lightAndShadowScene.addShadowTechnique( shadowMapAtlas );
+                shadowMapAtlas.setShadowSettings( shadowSettings );
+                this._shadowMapAtlas = shadowMapAtlas;
+
+
+                for ( var k = 0; k < numLights; k++ ) {
+                    this.addShadowedLight( group, k, lightScale );
+                }
+
+                var lightNumbers = [];
```

Comment:
this part can be done when traversing / applying shadowMapAtlas. In this case it's simpler to avoid the user to take care about this.